src/snowcli/cli/nativeapp/commands.py

```diff
@@ -90,20 +96,70 @@ def app_bundle(
 @with_project_definition("native_app")
 @global_options_with_connection
 def app_run(
+    version: Optional[str] = typer.Option(
+        None,
+        help=f"""The identifier or 'version string' of the version you would like to create a version and/or patch for.
+        Defaults to undefined, which means the CLI will use the version, if present, in the manifest.yml.""",
+    ),
+    patch: Optional[str] = typer.Option(
+        None,
+        "--patch",
+        help=f"""The patch number you would like to create for an existing version.
+        Defaults to undefined if it is not set, which means the CLI will either use the version, if present, in the manifest.yml,
+        or auto-generate the patch number.""",
+    ),
+    from_release_directive: Optional[bool] = typer.Option(
+        False,
+        "--from-release-directive",
+        help=f"""Flag to upgrade the application using a release directive that must already be set on the application package.""",
+        is_flag=True,
+    ),
+    interactive: Optional[bool] = typer.Option(
+        False,
+        "--interactive",
+        "-i",
+        help=f"""Defaults to False. Passing in --interactive/-i turns this to true, i.e. we will prompt you to confirm certain actions before the CLI executes them.
+        If not provided, the CLI will try to determine whether you are in an interactive mode.""",
+        is_flag=True,
+    ),
+    force: Optional[bool] = typer.Option(
+        False,
+        "--force",
+        help=f"""Defaults to False. Passing in --force turns this to True, i.e. we will implicitly respond “yes” to any prompts that come up.
+        This flag should be passed in if you are not in an interactive mode and want the command to succeed.""",
+        is_flag=True,
+    ),
     **options,
 ) -> CommandResult:
     """
-    Creates an application package in your Snowflake account, uploads code files to its stage,
-    then creates (or upgrades) a development-mode instance of that application. As a note, this
-    command does not accept role or warehouse overrides to your `config.toml` file, because your
+    Without any flags, this command creates an application package in your Snowflake account, uploads code files to its stage,
+    then creates (or upgrades) a development-mode instance of that application.
+    If passed in the version, patch or release directive flags, this command upgrades your existing application instance, or creates one if none exists. It does not create an application package in this scenario.
+    As a note, this command does not accept role or warehouse overrides to your `config.toml` file, because your
     native app definition in `snowflake.yml` or `snowflake.local.yml` is used for any overrides.
```

Comment:
Not necessarily odd - the intent here is to inform the user that what they specify under `native_app` definition in project.yml file is the final resolved role/db/warehouse etc. Any previous values specified either in `config.toml` or via CLI args e.g. `--role` are not used here because of the nature of native apps dealing with two separate objects, hence a singular `--role` and such don't allow for that distinction.
Would you like this to be taken out entirely and only be mentioned in the documentation?

Reply:
That's what I don't agree with. Any time a command line option is used, a reasonable user would expect it to override any config file. This decision seems somewhat confusing and I can't justify it.

Reply:
1. Removed it from here, and ack on documentation.
2. Erroring out/not having the role,db,wh and schema options at all for native app commands can be a separate stretch PR. I'll check with the Snow CLI team on this.

Reply:
Let's bring this up at the next team meating? Might have to be after the workshop. ACK on stretch PR.